src/rust/ide/src/executor/delay.rs

```diff
@@ -0,0 +1,82 @@
+//! Module defining `FrameDelay` - a `Future` that wraps a closure that should be called at a later
+//! time.
+
+pub use enso_prelude::*;
+
+use enso_protocol::prelude::Future;
+use ensogl_system_web::prelude::Closure;
+use ensogl_system_web::request_animation_frame;
+use futures::task::Context;
+use futures::task::Poll;
+use futures::task::Waker;
+use std::pin::Pin;
+
+
+// ========================
+// === Type Definitions ===
+// ========================
+
+type OnFrameClosure = Closure<dyn FnMut(f64)>;
+
+
+
+// ===================
+// === Frame Delay ===
+// ===================
+
+
+/// Delayed calls to a closure. The time delays is specified in number of render frames.
+#[derive(Debug)]
+pub struct FrameDelay<T>{
+    /// Counter that keeps track of the passage of frames. Needs to be shared with the JS runtime.
+    frames_left : Rc<RefCell<u64>>,
+    /// Internal closure that is shared with the JS runtime and updates the `frames_left`.
+    on_frame    : RefCell<Option<OnFrameClosure>>,
```

Comment:
Why we need to have RefCell here?

I think this structure can happily implement `Unpin` and thus `on_frame_action_init_if_uninit` be implemented on `&mut self`.